pkgs/fc/agent/fc/util/enc.py

```diff
@@ -30,13 +30,43 @@ def load_enc(log, enc_path):
     return enc
 
 
-def seed_enc(tmpdir, enc_path):
-    tmp_enc_path = tmpdir / "fc-data/enc.json"
-    if not enc_path.exists():
-        return
-    if not tmp_enc_path.exists():
+def seed_enc(log, tmpdir, enc_path):
+    """Populates ENC data during bootstrapping. Our automation puts an
+    initial ENC file in /tmp which provides the password to allow the agent to
+    talk to the directory and get updated ENC data from there."""
+    if enc_path.exists():
+        log.debug(
+            "seed-enc-present",
+            _replace_msg=(
+                "ENC file already present at {enc_path}, seeding not "
+                "required."
+            ),
+            enc_path=str(enc_path),
+        )
         return
-    shutil.move(tmp_enc_path, enc_path)
+
+    initial_enc_path = tmpdir / "fc-data/enc.json"
+    if initial_enc_path.exists():
+        log.info(
+            "seed-enc-seeding",
+            _replace_msg=(
+                "ENC file not found at {enc_path}, using initial data from "
+                "{initial_enc_path}"
+            ),
+            enc_path=str(enc_path),
+            initial_enc_path=str(initial_enc_path),
+        )
+        shutil.move(initial_enc_path, enc_path)
+    else:
+        log.info(
+            "seed-enc-initial-data-not-found",
+            _replace_msg=(
+                "ENC file not found at {enc_path} and initial data from "
```

Comment:
```suggestion
                "ENC file not found at {enc_path} and initial seed data from "
```

see terminology remark above